### DISCUSSION TITLE
H2 HD v1.1.1 and 1.1.0 Price of Loyalty - mission 8 - differences I noticed

### DISCUSSION THREAD

**Discussion by TLCD17** [ihhub/fheroes2#8969] [General]
Differences and changes I noticed in the last version of fh2 and earlier.
I installed the latest version of fheroes2 (1.1.1), deleted the records of the last mission (8th) in the Price of Loyalty campaign and started playing from this level in the new version.
I noticed some differences that were since versions 1.1.0 and 1.1.1 of fheroes 2 in the 8th mission.

In the new version, the computer is not very aggressive at the beginning and you can see that it panics, as if it doesn't know what to do, even when no one is attacking it.
However, this is only an illusion, because the computer is still strong and can gather a larger army than a human player.
The computer goes to the island less often than in the previous version, but this does not mean that it does not know when to go to defeat the human player.

Regarding the human player, I see that he has a higher gold gain than in version 1.1.0 and it is easier to gather more troops, but it is still not enough to buy more armies in castles captured to defeat the computer, which quickly gathers larger armies than a human player.

The computer has a surprisingly large army before the 3rd month.

<img width="1280" alt="Player and computer - army in three months" src="https://github.com/user-attachments/assets/3645b609-9def-47b6-a272-141aa1f4337d">

If anyone managed to complete this mission (8th) of the Price of Loyalty campaign in the latest version, please tell me how to do it. For me, this mission is still impossible to complete.

**Comment by Alucard648** [comment]
It`s even worse. Opponent has 3 fully built castles and 2 ultimate arts from start. And necromancer has another 2 fully built castles+ huge army and stats.

**Comment by fheroes2bugs** [comment]
There are already 2 discussions about mission 8 of the Price of Loyalty:
https://github.com/ihhub/fheroes2/discussions/8759
and
https://github.com/ihhub/fheroes2/discussions/8764

As I understand - at "impossible" game difficulty computer gets extra resources. Not sure, if it is fair to human, and not sure if it was the same way on original game...

At 1.1.1 I can see that should be fixed:
![image](https://github.com/user-attachments/assets/9dcffc0a-3796-4a84-ba0e-74fb7a8138e0)

Can't see any other reasons how computer can develop that fast. AI is still pretty stupid, especially on adventure map.

**Comment by ihhub** [comment]
Moving this topic to discussion section as this is not an issue.

**Comment by TLCD17** [comment]
I don't see any solution or idea for completing this mission. A computer is always better than a human - it has more troops out of nowhere.
In my opinion, if patches for heroes2 have been made available since the last version in terms of lowering the AI ​​difficulty level along with their resources, then the final release should be the earlier completion of this mission by people who tested the fheroes2 version before it was released.
Even if at some point in the 8th mission there is a hero hired in an inn who has reached level 20 (or got such a level at the beginning) with spells such as resurrection, dimension gates, mirror image, and various other useful spells, and our hero he is then about level 6, so even when controlling the hero mentioned above, with level 20 and good spells, he is not able to defeat each of his opponents due to the large army they have gathered. Besides, the human player's army is too small regardless of how he played this mission to defeat the AI.

**Comment by oleg-derevenetz** [comment]
I suggest to try to reduce the campaign difficulty by one step by selecting the Easy difficulty for this mission (which, considering the difficulty of the mission map itself, will correspond to the Hard AI difficulty):

![difficulty](https://github.com/user-attachments/assets/a2b7550c-c46c-4ba4-9585-f89a340bfbc5)

On this difficulty, AI has just one "virtual" extra Gold mine (+1000 Gold per day) as a resource bonus, that's all.

**Reply by fheroes2bugs** [reply]
> (+1000 Gold per day)

What bonuses have computer at Normal and Hard difficulty? The same, as is in the OG?

**Reply by oleg-derevenetz** [reply]
On Normal difficulty, there are no bonuses at all. On Hard difficulty, AI has +1000 Gold income per day, and no other bonuses (just limitations - e.g. it has a limit of 2 DD casts per day per hero on Normal difficulty and 3 DD casts per day per hero on Hard difficulty). See #8321 for details. OG has different set of bonuses.

**Comment by TLCD17** [comment]
In heroes 2 I always play on easy difficulty. The described hero in the tavern, who is level 20, and whom I bought, I do not know if anyone defeated him and ended up in the tavern. I think that the AI ​​with such a starting level or similar is unfair to the human player, who starts from the first level. Please investigate this matter.

<img width="1280" alt="Level 20 Hero in Tavern" src="https://github.com/user-attachments/assets/e7a3998d-1da5-4fca-95f4-830e0ada10ed">

<img width="1280" alt="Level 20 hero bought in a tavern" src="https://github.com/user-attachments/assets/fd913f8e-4618-4a6d-b026-da026126fe17">

**Comment by TLCD17** [comment]
I made screenshots at which point it is impossible to pass, even with the best possible hero (and not my own, but bought at the tavern) along with very good spells that cannot be achieved with the main hero at this stage of the game, which this screenshot shows. The human player's army is the largest I managed to get - there is no gold for a larger army, and the earlier losses are as small as possible.


<img width="1280" alt="Army - 1" src="https://github.com/user-attachments/assets/d6e542bc-b078-4ffe-8231-146011efadc9">

<img width="1280" alt="spells - 1" src="https://github.com/user-attachments/assets/e5654770-cf55-41d1-8a2e-1964835d1de1">

<img width="1280" alt="spells - 2" src="https://github.com/user-attachments/assets/3ea9006d-00cd-4bce-93b4-a55dfdc6b54c">

<img width="1280" alt="spells - 3" src="https://github.com/user-attachments/assets/808f38bc-c44a-47bd-931d-27f7c7b95e43">

**Comment by TLCD17** [comment]
I didn't forget to give the artifacts to the level 20 hero, but it doesn't make much difference. The AI ​​is stronger anyway.

<img width="1280" alt="Artifacts" src="https://github.com/user-attachments/assets/a48993e9-fc9a-4729-92f5-dd558c60f850">

**Comment by oleg-derevenetz** [comment]
>  I think that the AI ​​with such a starting level or similar is unfair to the human player, who starts from the first level.

As @Alucard648 mentioned earlier, at least Red player (Necromancer) has multiple heroes of this kind right from the start. Also AI on this mission has developed castles (Red's castles are fully built with level 5 Mage Guild, etc, other AI players also have less or more developed castles depending on player) and many resource piles on their territory. This is by design, that's how this map was created by the HoMM2 developers. In the original HoMM2, AI just didn't know how to do many things - e.g. it didn't know how to use Dimension Door or Summon Boat spells (which undoubtedly appear sooner or later with so many castles, if not from the very beginning). According to the developers of this map, AI (e.g. talking about Red player) should just sit on its island and wait for you. However, it doesn't work like that in fheroes2 anymore. In fheroes2 AI on these islands behaves actively - it uses Dimension Doors, summons/captures boats, conducts the map discovery and eventually finds you.

**Reply by zenseii** [reply]
I'd like to add a small correction. The OG AI did know how how to use Dimension Door, it just wasn't always very effective at it. This is also explained by the fact that in the OG the AI had the advantage of higher chance to get DD in its castles than the human player - another one of its cheats.

So with that said, I've heard many say that this particular map was harder in the OG than with fheroes2 due to the OG AI getting full knowledge of the map early on.

The only thing that could make this map more challenging in fheroes2 compared to the OG is that the new AI makes a champion hero, but really I've found this actually makes this map in particular easier because it is much easier to maneuver around one strong hero than several.

**Reply by oleg-derevenetz** [reply]
Hi @zenseii

> The OG AI did know how how to use Dimension Door

Yes, indeed, just checked it. In fact, I think that Summon Boat spell is even more "destructive" for the gameplay on a number of maps, because it was common practice to "lock up" a strong player on an island and not give him a boat, something like it's done with the Red player on this mission. Therefore, I suggest #8991. What do you think about that?

**Comment by TLCD17** [comment]
@oleg-derevenetz The conclusion is that campaigns and missions fheroes 2 is easier in the new version, as far as there is reprogrammed, and at the same time more difficult to defeat from the original version, due to the AI, which is more developed in the new version H2 HD.

**Comment by fheroes2bugs** [comment]
@TLCD17 @Alucard648 here is someone doing this map in fheroes: https://www.youtube.com/watch?v=83mwVae2y6g
don't know what version he is playing, but video is only 1 week old
